src/common/time/ESSTime.h

```diff
@@ -19,11 +19,85 @@ namespace esstime {
 using TimePointNano = std::chrono::high_resolution_clock::time_point;
 /// \brief Duration in nanoseconds with uint64_t precision
 using TimeDurationNano = std::chrono::duration<uint64_t, std::nano>;
+/// \brief Duration in microseconds with uint64_t precision
+using TimeDurationMicro = std::chrono::duration<double, std::micro>;
 /// \brief Duration in milliseconds with double precision
 using TimeDurationMilli = std::chrono::duration<double, std::milli>;
 /// \brief Duration in seconds with double precision
 using TimeDurationSec = std::chrono::duration<double, std::ratio<1>>;
 
+/// \brief Converts a frequency in Hertz to a period in nanoseconds.
+/// \param FrequencyHz Frequency in Hertz.
+/// \return std::chrono::nanoseconds The corresponding period in nanoseconds.
+inline std::chrono::nanoseconds hzToNanoseconds(const float &FrequencyHz) {
+  uint64_t nsPeriod = static_cast<uint64_t>(1e9 / FrequencyHz);
+  return std::chrono::nanoseconds(nsPeriod);
+}
```

Comment:
Passing simple types as const ref is a little overkill. Just pass by value like this

 ```suggestion
inline std::chrono::nanoseconds hzToNanoseconds(float FrequencyHz) {
  uint64_t nsPeriod = static_cast<uint64_t>(1e9 / FrequencyHz);
  return std::chrono::nanoseconds(nsPeriod);
}
```

I suggest you clean this up in all other similar places such as `nsToMicroseconds` etc.